src/wallet/Wallet.ts

```diff
@@ -1680,22 +1662,45 @@ class Wallet {
 	/**
 	 * @deprecated Use createMeltQuoteBolt11.
 	 */
-	async createMeltQuote(invoice: string): Promise<MeltQuoteBolt11Response> {
-		return this.createMeltQuoteBolt11(invoice);
+	async createMeltQuote(invoice: string, amountMsat?: number): Promise<MeltQuoteBolt11Response> {
+		return this.createMeltQuoteBolt11(invoice, amountMsat);
 	}
 
 	/**
 	 * Requests a melt quote from the mint. Response returns amount and fees for a given unit in order
 	 * to pay a Lightning invoice.
 	 *
 	 * @param invoice LN invoice that needs to get a fee estimate.
+	 * @param amountMsat Optional amount in millisatoshis to attach for amountless invoices, must not
+	 *   be provided for invoices that already encode an amount.
 	 * @returns The mint will create and return a melt quote for the invoice with an amount and fee
 	 *   reserve.
 	 */
-	async createMeltQuoteBolt11(invoice: string): Promise<MeltQuoteBolt11Response> {
+	async createMeltQuoteBolt11(
+		invoice: string,
+		amountMsat?: number,
+	): Promise<MeltQuoteBolt11Response> {
+		if (amountMsat != null && invoiceHasAmountInHRP(invoice)) {
+			throw new Error(
+				'amountMsat supplied but invoice already contains an amount, Leave amountMsat undefined for non-zero invoices.',
+			);
+		}
+
+		const supportsAmountless = this._mintInfo?.supportsAmountless?.('bolt11', this._unit) ?? false;
+
 		const meltQuotePayload: MeltQuoteBolt11Request = {
 			unit: this._unit,
 			request: invoice,
+
+			...(supportsAmountless && amountMsat
```

Comment:
If someone passes `amountMsat: 0`, it passes the null check in line 1683 but gets dropped in the spread (0 is falsy).